data-raw/healthy-places/travel-time-to-gp.R

```diff
@@ -0,0 +1,233 @@
+    current_locations <- apply(current_locations_df, 1, function(x)
+      make_location(id = x['id'], coords = list(lat = as.numeric(x["lat"]),
+                                                lng = as.numeric(x["lng"]))))
+    current_locations <- unlist(current_locations, recursive = FALSE)
+
+    current_search <-
+      make_search(
+        id = str_glue("search {current_sdz_centroid$id}"), # Make up an ID for the search so each search is unique
+        departure_location_id = current_sdz_centroid$id,
+        arrival_location_ids = as.list(current_GPs$id),
+        travel_time = 10800,  # 3 hours (in seconds)
+        properties = list("travel_time"),
+        arrival_time_period = "weekday_morning",
+        transportation = list(type = "public_transport")
+      )
+
+    current_result <- time_filter_fast(locations = current_locations, arrival_one_to_many = current_search)
+
+    # Convert JSON result to a data frame
+    current_result_df <- fromJSON(current_result$contentJSON, flatten = TRUE)
+
+    # Some Super Data Zones can't reach any GPs - ignore them
+    if (length(current_result_df$results$locations[[1]]) > 0) {
+      current_travel_time <-
+        current_result_df$results$locations[[1]] |>
+        as_tibble() |>
+        mutate(
+          # `travel_time` column is in seconds; convert to minutes
+          travel_time_mins = properties.travel_time / 60,
+          sdz21_code = current_sdz_centroid$id
+        ) |>
+        select(sdz21_code, osm_id = id, travel_time_mins)
+
+      GP_travel_time <- bind_rows(GP_travel_time, current_travel_time)
+    }
+
+    print(str_glue("Finished SDZ {sdz} of {nrow(current_sdz_centroids)}"))
+    Sys.sleep(2)
+  }
+
+  # Save progress to disc after each LAD
+  # NOTE: Please manually delete these files once the loop completes
+  write_csv(GP_travel_time, str_glue("data-raw/healthy-places/GP_travel_time-{i}.csv"))
+
+  print(str_glue("Finished Local Authority {i} of {nrow(ni_lad)}"))
+}
+
+GP_travel_time <- distinct(GP_travel_time)
+
+# Save the complete dataset for travel time from Intermediate Zones to GPs
+# This won't be available in the R package itself but want to keep it in the GitHub repo
+# since it takes quite a while to calculate
+write_csv(GP_travel_time, "data-raw/healthy-places/GP_travel_time.csv")
+
+GP_travel_time <- read_csv("data-raw/healthy-places/GP_travel_time.csv")
+
+# Look up Local Authorities for each Intermediate Zone and GP
+lookup_sdz_lad <-
+  lookup_dz21_sdz21_dea14_lgd14 |>
+  distinct(sdz21_code, ltla21_code = lgd14_code, ltla_name = lgd14_name)
+
+GP_travel_time <-
+  GP_travel_time |>
+  left_join(lookup_sdz_lad)
+
+# What are the fastest travel times within each Intermediate Zone?
+GP_travel_time_fastest <-
+  GP_travel_time |>
+  select(-osm_id) |>  # We don't need to know the GP ID for this
+  group_by(sdz21_code) |>
+  filter(travel_time_mins == min(travel_time_mins)) |>
+  ungroup() |>
+  distinct()
```

Comment:
Why do we calculate/use the fastest travel time here. Wouldn't something like the mean be better:
```
GP_travel_time |>
  select(-osm_id) |> 
  summarize(travel_time_mins = mean(travel_time_mins), .by = "sdz21_code")
```
Or alternatively, the slowest travel time, so those most in need are covered by the analysis?

Reply:
This is because the England Health Index [calculates average minimum distance](https://www.ons.gov.uk/peoplepopulationandcommunity/healthandsocialcare/healthandwellbeing/methodologies/healthindexindicatorsanddefinitions#healthy-places-access-to-services) to travel to the nearest GP/pharmacy/sports centre. Replicating that here, but with travel time instead of distance.

Reply:
I find the term "**average** minimum" unclear here. How can something be both an average and a minimum? Do we have access to any code or maths which shows how they calculate this?

I feel like the approach taken in this index isn't quite right. By taking the minimum, is this metric not sensitive to highly skewed left tails on the distribution of travel times? What do you think instead about taking the first quartile mean?

Reply:
I'm reading "average minimum" as the average of the minimum travel times/distances for the neighbourhoods within a Local Authority. But appreciate it's unclear. As far as I'm aware, there aren't any detailed technical docs nor code published anywhere.

I take your point about skewed tails. I'll explore the data a bit more this week and have a think about it.

Reply:
Looking through the [other indicators](https://www.ons.gov.uk/peoplepopulationandcommunity/healthandsocialcare/healthandwellbeing/methodologies/healthindexindicatorsanddefinitions) included in the Health Index, many of them are averages (e.g. crude death rate, percentage of people with (physical health condition), crude crime rates, etc.). For consistency, I'm now wondering whether we should take the mean travel time from neighbourhoods (Super Data Zones, in this case) in each Local Authority. What do you think @MikeJohnPage ?

The travel times from each Super Data Zone to GPs, pharmacies and sports centres are relatively normally distributed:
![image](https://github.com/user-attachments/assets/ca58290a-cc34-4038-9f56-c3ef1769a2b6)